### PR TITLE
Fix opensuse provisioning support

### PR DESCRIPTION
**Pull request by ryg-**
task/selinux.py, orchestra/remote.py: disable SELinux for opensuse/suse distros
nuke/actions.py: added support for sle/opensuse distros

Signed-off-by: Roman Grigorev <rgrigorev@suse.de>

### CODE REVIEW

**Comment by kshtsk** [comment]
ok to test

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by ryg-** [comment]
@susebot run deploy

**Comment by ryg-** [comment]
Ti be honest



> Would be nice to have a bit more of a description (in the commit msg) what is broken and **why** this is needed (see also https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#id7 )

commit message is updated

**Comment by kshtsk** [comment]
suse:tier0 in sepia for this PR http://pulpito.ceph.com/kyr-2019-12-09_19:46:45-suse:tier0-nautilus-none-basic-smithi/

File: teuthology/nuke/actions.py [hunk 1]
```diff
@@ -267,14 +268,26 @@ def remove_ceph_packages(ctx):
...
-            log.info('Remove any ceph packages')
-            remote.run(
-                args=['sudo', 'yum', 'remove', '-y', run.Raw(pkgs)],
-                check_status=False
+                args=['sudo', 'rpm', '--rebuilddb']
             )
+            if dist_release in ['opensuse', 'sle']:
+                remote.run(
+                    args=['sudo', 'zypper', 'clean']
+                )
```
Comment (smithfarm) [inline teuthology/nuke/actions.py]: any reason why `remote.run` instead of `remote.sh` ?

Comment (kshtsk) [reply teuthology/nuke/actions.py]: yeah... looks like copy-paster

**Comment by kshtsk** [comment]
> suse:tier0 in sepia for this PR http://pulpito.ceph.com/kyr-2019-12-09_19:46:45-suse:tier0-nautilus-none-basic-smithi/

@ryg- the run above has used following file:
```
$ cat nautilus-overrides.yaml 
overrides:
  install:
    ceph:
      packages:
        deb: []
        rpm: []
    repos:
    - name: nautilus
      url: https://download.opensuse.org/repositories/filesystems:/ceph:/nautilus/openSUSE_Leap_15.1/
    - name: distro
      url: http://download.opensuse.org/distribution/leap/15.1/repo/oss/
    - name: update
      url: http://download.opensuse.org/update/leap/15.1/oss/
```

And the command:
```
teuthology-suite -m smithi -k none --ceph-repo https://github.com/ceph/ceph --ceph nautilus --suite-repo https://github.com/suse/ceph --suite-branch ses6-downstream-commits --filter opensuse --suite suse:tier0 -d opensuse -D 15.1 -v $PWD/nautilus-overrides.yaml --newest 100 --limit 1 -t refs/pull/1369/merge
```

This I had to apply the patch from the #1373

File: teuthology/nuke/actions.py (hunk 1 above)
Comment (ryg-) [reply teuthology/nuke/actions.py]: should I replace it?

Comment (kshtsk) [reply teuthology/nuke/actions.py]: This is a new code, why not to make it humanized?

**Comment by kshtsk** [comment]
@susebot run deploy

File: teuthology/orchestra/remote.py [hunk 2]
```diff
@@ -292,7 +292,8 @@ def chcon(self, file_path, context):
         :param file_path: The path to the file
         :param context:   The SELinux context to be used
         """
-        if self.os.package_type != 'rpm':
+        if self.os.package_type != 'rpm' or\
```
Comment (kshtsk) [inline teuthology/orchestra/remote.py]: Space after `or`.

File: teuthology/nuke/actions.py (hunk 1 above)
Comment (kshtsk) [reply teuthology/nuke/actions.py]: Could you please use _remote.sh_ like this:
```
remote.sh('sudo zippier clean')
```

File: teuthology/task/selinux.py [hunk 3]
```diff
@@ -53,6 +53,9 @@ def filter_hosts(self):
             if remote.is_vm:
                 msg = "Excluding {host}: VMs are not yet supported"
                 log.info(msg.format(host=remote.shortname))
+            elif remote.os.name in ['opensuse', 'sle']:
+                msg = "Excluding {host}: opensuse, sle are not yet supported"
```
Comment (kshtsk) [inline teuthology/task/selinux.py]: maybe change to?
"Excluding {host}: SELinux is not supported for '{os}' os_type yet"

File: teuthology/nuke/actions.py (hunk 1 above)
Comment (ryg-) [reply teuthology/nuke/actions.py]: done

File: teuthology/orchestra/remote.py (hunk 2 above)
Comment (ryg-) [reply teuthology/orchestra/remote.py]: done

File: teuthology/task/selinux.py (hunk 3 above)
Comment (ryg-) [reply teuthology/task/selinux.py]: done

**Comment by toabctl** [comment]
@ryg- needs a rebase to fix the test failure `ERROR: Package 'setuptools' requires a different Python: 2.7.12 not in '>=3.5'`

**Comment by ryg-** [comment]
@susebot run deploy

**Comment by toabctl** [comment]
@susebot run deploy

**Comment by smithfarm** [comment]
Hi @ryg- This doesn't fix the kernel task, though, right? We still have to use `--kernel none` on opensuse.

File: teuthology/task/selinux.py [hunk 4]
```diff
@@ -53,6 +53,10 @@ def filter_hosts(self):
             if remote.is_vm:
                 msg = "Excluding {host}: VMs are not yet supported"
                 log.info(msg.format(host=remote.shortname))
+            elif remote.os.name in ['opensuse', 'sle']:
+                msg = "Excluding {host}: \
+                        SELinux is not supported for '{os}' os_type yet"
+                log.info(msg.format(host=remote.shortname))
```
Comment (smithfarm) [inline teuthology/task/selinux.py]: `os=remote.os.name` missing here, causing test failure:

```
2020-01-17T20:50:22.461 INFO:teuthology.run_tasks:Running task selinux...
2020-01-17T20:50:22.482 ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_wip-fix-opensuse-provisioning/teuthology/run_tasks.py", line 86, in run_tasks
    manager = run_one_task(taskname, ctx=ctx, config=config)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_wip-fix-opensuse-provisioning/teuthology/run_tasks.py", line 65, in run_one_task
    return task(**kwargs)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_wip-fix-opensuse-provisioning/teuthology/task/selinux.py", line 42, in __init__
    super(SELinux, self).__init__(ctx, config)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_wip-fix-opensuse-provisioning/teuthology/task/__init__.py", line 36, in __init__
    self.filter_hosts()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_wip-fix-opensuse-provisioning/teuthology/task/selinux.py", line 59, in filter_hosts
    log.info(msg.format(host=remote.shortname))
KeyError: 'os'
```

**Comment by smithfarm** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by ryg-** [comment]
run in sepia started in  01/19
http://pulpito.ceph.com/ryg--2020-01-19_15:38:30-suse:tier0-nautilus-none-basic-smithi/

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@susebot run deploy

**Pull request merged**